blablahelp-backend/src/test/java/org/nnn4eu/hfische/blablahelp/blablahelpbackend/userdata/web/UserDataCtrlTest.java

```diff
@@ -0,0 +1,180 @@
+import org.nnn4eu.hfische.blablahelp.blablahelpbackend.shared.model.EAddressType;
+import org.nnn4eu.hfische.blablahelp.blablahelpbackend.userdata.UserDataService;
+import org.nnn4eu.hfische.blablahelp.blablahelpbackend.userdata.model.Offer;
+import org.nnn4eu.hfische.blablahelp.blablahelpbackend.userdata.model.UserData;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+import java.lang.reflect.Type;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.csrf;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.user;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+class UserDataCtrlTest {
+
+    private Account account=createAccount();
+    private final String id= "97a6c939-0919-44f5-99b3-1df1543c7427";
+    @Autowired
+    private MockMvc mockMvc;
+    @Autowired
+    ObjectMapper objectMapper;
+
+    @Autowired
+    UserDataService userDataService;
+    @Autowired
+    AccountService accountService;
+
+    @WithMockUser(authorities = {"BASIC"})
+    @DirtiesContext
+    @Test
+    void getUserData_throw() throws Exception {
+        UserData userData=new UserData(id);
+        mockMvc.perform(
+                        get(UrlMapping.USERDATA+"/"+id))
+                .andExpect(status().isConflict());
+    }
+
+
+    @WithMockUser(authorities = {"BASIC"})
+    @DirtiesContext
+    @Test
+    void getUserData() throws Exception {
+        String accountId= account.getId();
+        accountService.saveNew(account);
+        UserData userData=new UserData(accountId);
+        userData.setVersion(0L);
+        mockMvc.perform(
+                get(UrlMapping.USERDATA+"/"+accountId))
+                .andExpect(status().isOk())
+                .andExpect(content().json(objectMapper.writeValueAsString(userData)));
+
+        Offer offer=createOffer(accountId);
+        offer=createNewOfferTest(offer);
+
```

Comment:
😂
I know, we did this - but using tests to test a test is ..mmhhhh... what should I say... I am sure, you know what I mean - letting you know, I would do the same to save time... 🤷🏻‍♀️🤭